### DISCUSSION TITLE
Upcoming 1.1.0

### DISCUSSION THREAD

**Discussion by irvinm** [irvinm/TST-Colorize-Tabs#1] [Announcements]
Changelog so far ...

----- 

- Forked version from original addon
- Initial version attempting to fix issue of losing color information between restarts
  - https://github.com/emvaized/tst-colorize-tabs/issues/8
- Converted browser.storage.local.set\get to browser.sessions.getTabValue\setTabValue
- Removed unneeded index code
- Added addon icon - Attribution to Flaticon creator Hasymi
- Added support for badge count for number of tabs with colors
- Removed customization of menu enable\disable. 
  - Felt it was confusing especially with multi-select of existing multi-colored tabs.
- Added support for keyboard shortcuts to move up or down to colorized tabs
  - Default is "Alt+Down for next" and "Alt+Up for previous"
  - Can change shortcuts via about:addons -> cogwheel -> Manage Extension Shortcuts
  - https://support.mozilla.org/en-US/kb/manage-extension-shortcuts-firefox
- https://github.com/irvinm/TST-Colorize-Tabs/issues/3
  - Default shortcuts are ALT+(1-8)
  - ALT+(1-7) for each color and ALT+8 to remove all colors
  - Should work for individual tabs or all tabs that are currently selected
- https://github.com/irvinm/TST-Colorize-Tabs/issues/2
  - New options for light\dark options page
  - New options for keyboard behavior to stop at top\bottom or cycle
  - Options can be seen by clicking on addon icon or about:addons -> Options

**Comment by Nalla22** [comment]
Hello, thank you very much for this update, it's a great job!

You've managed to combine the two extensions "tst-colorize-tabs & tst-tab-favorites" into a single extension, while solving the problems of both extensions. I'm very happy to use it 😃

**Comment by collector-ynh** [comment]
Hello, I've just tested the extension, it works magnificently well, it's a very good job 👍

So do the shortcuts, but I'd like to suggest two improvements that tst-tab-favorites already has:

1) Allow the user to assign a keyboard shortcut for each color.

2) If the user scrolls through the colored tabs with ALT+DOWN and passes the last tab, scrolling should continue in a circle and return to the first colored tab at the top. Conversely, using ALT+UP

I think these would be the last improvements before we bury tst-tab-favorites for good.

**Reply by irvinm** [reply]
For traversing the colored tabs, maybe I could add an option to stop at the top or bottom or "cycle" around.  I personally like stopping at the top or bottom, but I could see people wanting it to cycle as well.

Just to be clear on the other, you mean a shortcut for each color that can apply that color to the active tab?

**Reply by irvinm** [reply]
- https://github.com/irvinm/TST-Colorize-Tabs/issues/2
- https://github.com/irvinm/TST-Colorize-Tabs/issues/3

**Reply by Nalla22** [reply]
Indeed, it would be even better if it were possible to switch between a "cycle" and "stop on top and bottom" mode, giving the user a choice.

Yes, exactly, I was talking about giving the user the option of assigning shortcuts to each color that can be assigned to the active tab, as tst-tab-favorites already does. This would make it possible to avoid the drop-down menu and assign colors directly via shortcuts!

**Reply by irvinm** [reply]
I did both items today, but would like some others to try it out to see if there are any additional issues.  I've updated the info above.

**Reply by Nalla22** [reply]
Thank you very much for your work. I'll test it and if there's a problem, I'll report it here 🙂
